README.md

```diff
@@ -8,34 +8,318 @@ or
+
+`gem 'teepublic-api', github: 'bustedtees/teepublic-api'`
+
+and run
+
+$`bundle install`
+
+## Step 2 - Configure
+Add `config/initializers/tee_public.rb`
+and copy paste the content below with your api key
+```
+require 'tee_public/api'
+
+TeePublic::Api.configure do |config|
+  config.api_key = 'YOUR_API_KEY_GOES_HERE'
+end
+```
+
+## Step 3 - Explore APIs
+Open rails console `rails c` and try these
+#### List of Stores API
+```
+  fetch_stores = TeePublic::Api.send('stores')
+  stores = fetch_stores.dig('body', '_embedded', 'stores').map(&:deep_symbolize_keys!)
+  puts stores
+```
+
+#### Single Store API
+```
+  # store_id -> Pick a store_id returned by List of Stores API
+  # page -> For trial purposes, you can use page = 1
+  store_url = "stores/#{store_id}?page=#{page}"
+
+  # album_id -> it's an additional filter option, can keep it blank for now,
+  # we will learn more about this later on
+  store_url += "&album_id=#{album_id}" if album_id.present?
+
+  # product_type -> same explanation as album_id
+  store_url += "&product_type=#{product_type}" if product_type.present?
+
+  fetch_store = TeePublic::Api.send(store_url).deep_symbolize_keys!    
+  store = fetch_store[:body]
+  puts store
+```
+
+#### Design API
+If no designs are found in Single Store API that means you have not added any designs to that store on teepublic.com. Head to TeePublic.com, locate that store from slug, '/stores/<store-slug>'. Add few designs to it.
+```
+  # design_id -> Single Store API returns a list of designs. Pick any design_id you like
+  fetch_design = TeePublic::Api.designs(design_id).deep_symbolize_keys!
+  design = fetch_design[:body]
+  puts design
+```
+
+#### SKUs API
+Each design can be printed on different products like t-shirt, mug or case. Further more, each product can have different skus like t-shirt colors, sizes, etc. Skus are different variants of a product for a given design
+```
+  # design_id -> Single Store API returns a list of designs. Pick any design_id you like
+  # product_type -> Design API returns a list of skus, sku object has a key called "productType"
+  skus_url = "designs/#{design_id}/products/#{product_type}"
+  fetch_skus = TeePublic::Api.send(skus_url).deep_symbolize_keys!
+  skus = fetch_skus[:body]
+  puts skus
+```
+
+#### Affiliate Network ID API
+Affiliate Network ID is a unique ID assigned to an application (like https://mywebsite.com) on TeePublic system. This is required by TeePublic during checkout to make sure it credit sales back to an affiliate network.
+```
+# store_url is explained in "Single Store API" section
+fetch_store = TeePublic::Api.send(store_url).deep_symbolize_keys!
+affiliate_network_id = fetch_store.dig(:headers, :'x-affiliatenetwork-id')
+puts affiliate_network_id
+```
+
+## Step 4 -  Add Routes to routes.rb file
+Application will need at-least below three routes. One for teepublic store, one for teepublic design and one for cart page.
```

Comment:
Needs cleanup.  "Will need the 3 routes  below:  the storefront, the design, and the cart."

Reply:
Cool